cvat-core/src/requests-manager.ts

```diff
@@ -0,0 +1,357 @@
+        this.#progress = initialData.progress;
+        this.#message = initialData.message;
+        this.#resultUrl = initialData.result_url;
+
+        this.#enqueuedDate = initialData.enqueued_date;
+        this.#startedDate = initialData.started_date;
+        this.#finishedDate = initialData.finished_date;
+        this.#expiryDate = initialData.expiry_date;
+
+        if (initialData.owner) {
+            this.#owner = new User(initialData.owner);
+        }
+    }
+
+    get id(): string {
+        return this.#id;
+    }
+
+    get status(): RQStatus {
+        return this.#status.toLowerCase() as RQStatus;
+    }
+    set status(status) {
+        this.#status = status;
+    }
+
+    get progress(): number {
+        return this.#progress;
+    }
+
+    get message(): string {
+        return this.#message;
+    }
+    set message(message) {
+        this.#message = message;
+    }
+
+    get operation(): Operation {
+        return {
+            target: this.#operation.target,
+            type: this.#operation.type,
+            format: this.#operation.format,
+            jobID: this.#operation.job_id,
+            taskID: this.#operation.task_id,
+            projectID: this.#operation.project_id,
+        };
+    }
+
+    get url(): string {
+        return this.#resultUrl;
+    }
+
+    get enqueuedDate(): string {
+        return this.#enqueuedDate;
+    }
+
+    get startedDate(): string {
+        return this.#startedDate;
+    }
+
+    get finishedDate(): string {
+        return this.#finishedDate;
+    }
+
+    get expiryDate(): string {
+        return this.#expiryDate;
+    }
+
+    get owner(): User {
+        return this.#owner;
+    }
+
+    updateFields(request: Partial<SerializedRequest>): void {
+        if (request.id !== undefined) {
+            this.#id = request.id;
+        }
+        if (request.status !== undefined) {
+            this.#status = request.status as RQStatus;
+        }
+        if (request.operation !== undefined) {
+            this.#operation = request.operation;
+        }
+        if (request.progress !== undefined) {
+            this.#progress = request.progress;
+        }
+        if (request.message !== undefined) {
+            this.#message = request.message;
+        }
+        if (request.result_url !== undefined) {
+            this.#resultUrl = request.result_url;
+        }
+        if (request.enqueued_date !== undefined) {
+            this.#enqueuedDate = request.enqueued_date;
+        }
+        if (request.started_date !== undefined) {
+            this.#startedDate = request.started_date;
+        }
+        if (request.finished_date !== undefined) {
+            this.#finishedDate = request.finished_date;
+        }
+        if (request.expiry_date !== undefined) {
+            this.#expiryDate = request.expiry_date;
+        }
+        if (request.owner !== undefined) {
+            this.#owner = new User(request.owner);
+        }
+    }
+}
```

Comment:
Why do not implement setters? The method makes sense to reduce code duplication, but in this case another question:

- We have setters for `status` and `message`, but not for others. Why?